FirebaseTestProject/Views/PostRow.swift

```diff
@@ -40,6 +43,24 @@ struct PostRow: View {
         }
     }
     
+    private var image: some View {
+        AsyncImage(url: URL(string: post.imageURL)) { image in
+            image
+                .resizable()
+                .aspectRatio(contentMode: .fit)
+                .clipShape(RoundedRectangle(cornerRadius: 10))
+                .frame(width: 300, height: 200)
+        } placeholder: {
+            VStack {
+                ProgressView()
+                Text("Loading Image")
+                    .font(.caption)
+            }
+            .frame(width: 300, height: 200)
+        }
+        .padding(.horizontal)
+    }
+    
     private var footer: some View {
```

Comment:
Same for the footer

Reply:
For this one, I wasn’t sure about moving the entire footer to a separate struct (a footer struct would require 3-4 parameters and be tightly coupled to the parent `PostRow`, which isn’t ideal). Instead, I separated out the `FavoriteButton` and `DeleteButton`—including their associated `TaskViewModel`s, alerts, and confirmation dialogs—and kept the remainder in the view `body`. Not sure if this was the best way to do it. If you could take a look and let me know what you think, I would really appreciate it!